src/components/Signup.jsx

```diff
@@ -1,95 +1,101 @@
-import { useEffect, useState } from "react";
-import svgImage from "../assets/signup.svg";
-import { useDispatch, useSelector } from "react-redux";
-import { registerUser } from "../redux/actions/authActions";
-import { useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react"
+import svgImage from "../assets/signup.svg"
+import { useDispatch, useSelector } from "react-redux"
+import { registerUser } from "../redux/actions/authActions"
+import { useNavigate } from "react-router-dom"
 
-import PropTypes from "prop-types";
+import PropTypes from "prop-types"
 
 const SignUp = ({ setSignin }) => {
-  const dispatch = useDispatch();
-  const navigate = useNavigate();
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
 
-  const {error, user, isAuthenticated} = useSelector(state => state.auth);
+  const { error, user, isAuthenticated } = useSelector((state) => state.auth)
 
   useEffect(() => {
-    if(isAuthenticated){
-      navigate('/dashboard');
+    if (isAuthenticated) {
+      navigate("/location")
     }
-  },[isAuthenticated]);
+  }, [isAuthenticated])
 
   const [formData, setFormData] = useState({
     username: "",
     email: "",
     password: "",
     confirm_password: "",
-  });
+  })
 
-  const [errors, setErrors] = useState({});
+  const [errors, setErrors] = useState({})
 
   const handleChange = (e) => {
-    const { name, value } = e.target;
+    const { name, value } = e.target
     setFormData({
       ...formData,
       [name]: value,
-    });
-  };
+    })
+  }
 
   const handleForm = (e) => {
-    e.preventDefault();
-    const validationErrors = {};
-
-
+    e.preventDefault()
+    const validationErrors = {}
     if (!formData.username.trim()) {
-      validationErrors.username = "Username is required";
+      validationErrors.username = "Username is required"
     } else if (!/^[a-zA-Z0-9]+$/.test(formData.username)) {
-      validationErrors.username = "Username must be alphanumeric";
+      validationErrors.username = "Username must be alphanumeric"
     }
     if (!formData.email.trim()) {
-      validationErrors.email = "Email is required";
+      validationErrors.email = "Email is required"
     } else if (
       !/^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/.test(formData.email)
```

Comment:
Try to keep all validation regexs in a separate file as constants. Helps in maintaining the code better.